### PR TITLE
Implement quick updates

### LINKED ISSUE #40: Rust-based app generator
For version 4 of the app.yml format, we should use a new app.ml -> docker-compose.yml converter written in Rust.

This means we will have a cleaner and faster converter. In addition, it can run in Docker so we can reduce the amount of host dependencies (So we can drop most python packages needed right now).

Having that in Docker will also make it easier to update it (outside of full updates).

App.yml v4 should also be more similar to docker-compose.yml to make it easier to port apps.

### PR DESCRIPTION
**Pull request by AaronDewes**
This contains:

- App.yml v4 (Fixes #40)
- Quicker updates
- Fixes to set-update-channel

### CODE REVIEW

File: app-system/sources.list [hunk 1]
```diff
@@ -3,9 +3,9 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
 
 # A collection of fully FLOSS app definitions and FLOSS apps for Citadel.
-https://github.com/runcitadel/apps v3-beta
+https://github.com/runcitadel/apps v3-stable
```
Comment (WilliamConnatser) [inline app-system/sources.list]: Did you mean to make this change? It seems curious, lol.. just making sure!

File: app-system/sources.list [hunk 2]
```diff
@@ -3,9 +3,9 @@
...
 
 # Some apps modified version of Umbrel apps, and their app definitions aren't FLOSS yet.
 # Include them anyway, but as a separate repo.
 # Add a # to the line below to disable the repo and only use FLOSS apps.
-https://github.com/runcitadel/apps-nonfree v3-beta
+https://github.com/runcitadel/apps-nonfree v3-stable
```
Comment (WilliamConnatser) [inline app-system/sources.list]: Same question here

**Review by WilliamConnatser — CHANGES_REQUESTED** [review]
Let me know about the question I had, and in the meantime I will try to test this. Not sure how I can spin up a Citadel node on the previous Citadel version so I can upgrade. I may have to enhance `citadel-dev` with this functionality? I am going to look into that momentarily.

File: app-system/sources.list (hunk 1 above)
Comment (AaronDewes) [reply app-system/sources.list]: No, this was a mistake, it should be `v4-dev` (and later `v4-beta` and `v4-stable`)

File: app/lib/manage.py [hunk 3]
```diff
@@ -49,26 +73,58 @@ def joinThreads(threads: List[threading.Thread]):
...
 def getArguments():
     arguments = ""
     for i in range(3, len(argv)):
         arguments += argv[i] + " "
     return arguments
 
+def handleAppV4(app):
+    composeFile = os.path.join(appsDir, app, "docker-compose.yml")
+    os.chown(os.path.join(appsDir, app), 1000, 1000)
+    os.system("docker run --rm -v {}:/apps -u 1000:1000 {} /app-cli convert --app-name '{}' --port-map /apps/ports.json /apps/{}/app.yml /apps/{}/result.yml --services 'lnd'".format(appsDir, dependencies['app-cli'], app, app, app))
```
Comment (pwltr) [inline app/lib/manage.py:90]: This pulls the image for the `app-cli` and runs containers for every v4 app I believe. It works but might lead to conflicts, or at least there's room for a speed improvement.

```
Unable to find image 'ghcr.io/runcitadel/app-cli:main@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054' locally
Unable to find image 'ghcr.io/runcitadel/app-cli:main@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054' locally
Unable to find image 'ghcr.io/runcitadel/app-cli:main@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054' locally
Unable to find image 'ghcr.io/runcitadel/app-cli:main@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054' locally
Unable to find image 'ghcr.io/runcitadel/app-cli:main@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054' locally
Unable to find image 'ghcr.io/runcitadel/app-cli:main@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054' locally
Unable to find image 'ghcr.io/runcitadel/app-cli:main@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054' locally
Unable to find image 'ghcr.io/runcitadel/app-cli:main@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054' locally
ghcr.io/runcitadel/app-cli@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054: Pulling from runcitadel/app-cli
ghcr.io/runcitadel/app-cli@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054: Pulling from runcitadel/app-cli
ghcr.io/runcitadel/app-cli@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054: Pulling from runcitadel/app-cli
ghcr.io/runcitadel/app-cli@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054: Pulling from runcitadel/app-cli
ghcr.io/runcitadel/app-cli@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054: Pulling from runcitadel/app-cli
ghcr.io/runcitadel/app-cli@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054: Pulling from runcitadel/app-cli
ghcr.io/runcitadel/app-cli@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054: Pulling from runcitadel/app-cli
ghcr.io/runcitadel/app-cli@sha256:f532923eac28cfac03579cbb440397bcf16c8730f291b39eeada8278331f7054: Pulling from runcitadel/app-cli
```

**Review by pwltr — CHANGES_REQUESTED** [review]

**Review by pwltr — APPROVED** [review]
Looks good. Only notable thing was this warning:

"WARNING: App 'ln-visualizer' has unknown dependency 'lightning'"

File: app-system/sources.list [hunk 4]
```diff
@@ -3,7 +3,7 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
 
 # A collection of fully FLOSS app definitions and FLOSS apps for Citadel.
-https://github.com/runcitadel/apps v3-beta
+https://github.com/runcitadel/apps v4-dev
```
Comment (WilliamConnatser) [inline app-system/sources.list:6]: Make sure to change this back

File: app/lib/manage.py [hunk 5]
```diff
@@ -49,26 +73,58 @@ def joinThreads(threads: List[threading.Thread]):
...
         sourceMap = json.load(f)
     if not name in sourceMap:
-        print("Warning: App {} is not in the source map".format(name))
+        print("Warning: App {} is not in the source map".format(name), file=sys.stderr)
         sourceMap = {
             name: {
-                "githubRepo": "runcitadel/core",
-                "branch": "v2"
+                "githubRepo": "runcitadel/apps",
+                "branch": "v4-stable"
```
Comment (WilliamConnatser) [inline app/lib/manage.py:127]: I am doubtful that this should be hard coded

File: app-system/sources.list (hunk 4 above)
Comment (WilliamConnatser) [inline app-system/sources.list]: Make sure to change this back when merging

**Review by WilliamConnatser — CHANGES_REQUESTED** [review]

File: app/lib/manage.py (hunk 5 above)
Comment (AaronDewes) [reply app/lib/manage.py:127]: This is just a fallback for apps that aren't found if people try to download them, it's basically unused (Which is why this was broken for so long, it's nearly unused). It should be removed though but that's for another PR.

File: app-system/sources.list (hunk 4 above)
Comment (AaronDewes) [reply app-system/sources.list]: Done

File: app/lib/manage.py (hunk 5 above)
Comment (WilliamConnatser) [reply app/lib/manage.py:127]: Cool cool, thanks for explaining. LGTM!

**Review by WilliamConnatser — APPROVED** [review]

**Pull request merged**